bindings/rust/standard/s2n-tls-hyper/Cargo.toml

```diff
@@ -13,8 +13,8 @@ publish = false
 default = []
 
 [dependencies]
-s2n-tls = { version = "=0.3.9", path = "../s2n-tls" }
-s2n-tls-tokio = { version = "=0.3.9", path = "../s2n-tls-tokio" }
+s2n-tls = { version = "=0.3", path = "../../s2n-tls" }
+s2n-tls-tokio = { version = "=0.3", path = "../../s2n-tls-tokio" }
```

Comment:
We keep the patch version when specifying s2n-tls dependencies in the other crates. Is there any reason for this?
https://github.com/aws/s2n-tls/blob/27e2fad08e098500c4b7f303be13877644089084/bindings/rust/s2n-tls-tokio/Cargo.toml#L18

Reply:
I think we could keep the patch version for now. In the future, we can try to do a [`workspace dependency`](https://github.com/aws/s2n-quic/pull/2400#discussion_r1868518216) in https://github.com/aws/s2n-quic/pull/2400 which saves time updating the version.

Reply:
Recapping some offline discussion. It's an easy way to avoid breaking versions when we add new apis to s2n-tls.

This is often necessary because we
1. add a new API to s2n-tls (C)
2. which adds a new symbol to `s2n-tls-sys` - e.g. 0.3.8`
3. if the `s2n-tls` crate references that symbol, then it needs >= 0.3.8 of `s2n-tls-sys` to build, and could not build with 0.3.1, for example.